mobile-app/app/screens/AppNavigator/screens/Loans/hooks/CollateralizationRatio.ts

```diff
@@ -54,3 +54,100 @@ export function useCollateralizationRatioColor (props: CollateralizationRatioPro
   }
   return style
 }
+
+export function useResultingCollateralizationRatioByCollateral ({
+  collateralValue,
+  collateralRatio,
+  minCollateralRatio,
+  totalLoanAmount,
+  numOfColorBars = 6,
+  totalCollateralValueInUSD
+}: {
+  collateralValue: string
+  collateralRatio: BigNumber
+  minCollateralRatio: BigNumber
+  totalLoanAmount: BigNumber
+  totalCollateralValue?: BigNumber
+  numOfColorBars: number
+  totalCollateralValueInUSD: BigNumber
+}): {
+  resultingColRatio: BigNumber
+  displayedColorBars: number
+} {
+  const hasCollateralRatio = !new BigNumber(collateralRatio).isNaN() && new BigNumber(collateralRatio).isPositive()
+  const resultingColRatio = collateralValue === '' || !hasCollateralRatio
+  ? new BigNumber(collateralRatio)
+  : totalCollateralValueInUSD.dividedBy(totalLoanAmount).multipliedBy(100)
```

Comment:
for calculating any forecast col ratio, we need to take into account of the interestPerBlock since the action of adding collateral will take one block.

Can refer to useResultingCollateralRatio https://github.com/DeFiCh/wallet/blob/8dbf007fdfd9562f907054c792b0feeaa62c2705/mobile-app/app/screens/AppNavigator/screens/Loans/hooks/CollateralPrice.ts#L25 in which we are using it to calculate the forecast col ratio when user change the amount of loan

I believe now the ratio you calculated vs the one returned from API is slightly off, try to see adding this would help

Reply:
To test @pierregee, you can add an E2E that the resulting col ratio displayed here will be equal to the col ratio of the vault card

Reply:
@kyleleow  I am looking into the interestPerBlock logic. In Borrowing tokens, interestPerBlock is multiplied with newLoanAmount that changed.
 
1. When borrowing token, I see a slight difference in the resulting colRatio vs api’s vault collateralRatio (.xx), Same as what’s happening on add/remove collateral
 2. Were you able to test the difference when adding/removing collateral? Hmm maybe I’m missing something. Also, do we need to take into account the interestPerBlock when adding/removing a collateral — I think it’s just for loaning tokens

@thedoublejay sure looking into it